crates/orderbook/src/database/orders.rs

```diff
@@ -315,7 +315,11 @@ fn calculate_status(order: &FullOrder) -> OrderStatus {
     if order.invalidated {
         return OrderStatus::Cancelled;
     }
-    if order.valid_to < Utc::now().timestamp() {
+    let now = Utc::now().timestamp();
+    if order.valid_to < now {
+        return OrderStatus::Expired;
+    }
+    if matches!(&order.ethflow_data, Some((_,valid_to)) if *valid_to < now) {
```

Comment:
Actually... maybe instead of doing that logic here, it makes sense to add an:
```rust
impl Order {
    fn actual_valid_to(&self) -> DateTime<Utc> {
        ...
    }
}
```

Again, just a suggestion, but I am OK with merging as is.